src/md/compatibility.md

```diff
@@ -0,0 +1,112 @@
+* PATCH version when you make backwards-compatible bug fixes.
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+The overall goal is to avoid backward incompatible changes and major release upgrades. Accordingly we attempt to release new features with minor versions that are incremental to the prior release and offer our users a frictionless upgrade path. When planning contributions, please consider compatibility and release road map upfront. Specifically, certain changes that conflict with the versioning may need to be documented in JIRA and deferred until a future major release. 
+
+##Compatibility types
+
+###Java API
+
+Public API compatibility is required to ensure end-user programs and downstream projects continue to work without modification.
+The public API consists of:
+
+* apex-core: all interfaces and classes in `api` and `common` modules
+* apex-malhar: all interfaces and classes in all modules except `demos`, `samples`, `benchmark` 
+
+Interfaces and classes that are part of the public API and are annotated with [interface stability](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/InterfaceClassification.html) are treated according to the rules defined by the annotation.  
+
+Policy
+
+Changes to the public API must follow semantic versioning. 
+Public APIs must be deprecated for at least one major release prior to their removal in a major release.
+The japicmp plugin is used to enforce compatibility as part of the Travis pre-commit builds.
+
+###Semantic compatibility
+
+The behavior of APIs needs to remain consistent over versions, though changes for correctness may result in changes in behavior. Tests and javadocs specify the behavior. Over time, test suites should be expanded to verify compliance with the specification, effectively creating a formal specification for the subset of behaviors that can be easily tested.
+
+Policy
+
+The behavior of API may be changed to fix incorrect behavior, changes to be accompanied by tests coverage for the exact behavior.
+
+###REST API
+
+REST API compatibility corresponds to both the URLs and request/response content over the wire. REST APIs are specifically meant for stable use by clients across releases, even major releases. 
+
+Policy
+
+The REST API is separately versioned. This is to allow for co-existence of old and new API should there be a need for backward incompatible changes in the future.
+
+###Command Line Interface (CLI)
+
+The CLI may be used either directly via the system shell or via shell scripts. Changing the path, removing or renaming command line options, the order of arguments, or the command return code and output break compatibility and may adversely affect users.
+
+Policy
+
+CLI commands are to be deprecated (warning when used) in a prior minor release before they are removed or incompatibly modified in a subsequent major release.
```

Comment:
For API the policy mentioned above is
"Public APIs must be deprecated for at least one major release prior to their removal in a major release."
For CLI the policy mentioned is
"CLI commands are to be deprecated (warning when used) in a prior minor release before they are removed or incompatibly modified in a subsequent major release."

I read this as keeping old APIs for an extra major release in a deprecated capacity, so for example if it is deprecated towards the end of the 3.x lifecycle it gets removed in 5.x and not 4.x. Looks like that is not what you are intending to say, then can we make the wording of the two statements above similar so as to avoid confusion.